plugins/kotlin/idea/tests/test/org/jetbrains/kotlin/idea/editor/quickDoc/PackageQuickDocTest.kt

```diff
@@ -0,0 +1,29 @@
+// Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.idea.editor.quickDoc
+
+import com.intellij.codeInsight.javadoc.JavaDocInfoGenerator
+import com.intellij.psi.PsiPackage
+import org.jetbrains.kotlin.idea.test.KotlinLightCodeInsightFixtureTestCase
+import org.junit.Assert
+
+class PackageQuickDocTest: KotlinLightCodeInsightFixtureTestCase() {
+    fun testNoDuplicates() {
+        val psiFile = myFixture.addFileToProject(
+            "com/example/nodupes/Fantastic.kt", """
+            package com.example.nod<caret>upes
+            class Fantastic
+        """.trimIndent()
+        )
+        myFixture.configureFromExistingVirtualFile(psiFile.virtualFile)
+
+        val psiPackage = myFixture.elementAtCaret as PsiPackage
+
+        val stringBuilder = StringBuilder()
+        JavaDocInfoGenerator(myFixture.project, psiPackage).generateDocInfoCore(stringBuilder, false)
+        val html = stringBuilder.toString()
+
+        val hrefPattern = Regex(Regex.escape("psi_element://com.example.nodupes.Fantastic"))
+        val numMatches = hrefPattern.findAll(html).count()
+        Assert.assertEquals(1, numMatches)
```

Comment:
it would be better to check not just the number of elements, but what exact elements they are:

when test fails, it gives more useful information about the problem.

Reply:
I agree it would be better to have more robust testing for quick documentation.

This test case, however, is only testing that there are no duplicates (`testNoDuplicates`) - if the test fails, it will be clear that there is a problem with duplicates. IMO, if we want to test a different aspect of the quickdoc, (e.g. that all classes are represented), that seems like a different test case outside the scope of this change, which is specifically to address the duplicate issue. 

There is a more comprehensive unit test for the changes I made to `JavaPsiFacadeImpl` in the other test I added.